sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -252,12 +260,12 @@ def sleep(self):
         sleep(self._statement_sleep_seconds)
 
     def refresh_status(self):
-        status = self._http_client.get_session(self.id)[u'state']
+        response = self._http_client.get_session(self.id)
+        status = response[u'state']
+        log_array = response[u'log']
 
-        if status in constants.POSSIBLE_SESSION_STATUS:
-            self.status = status
-        else:
-            raise LivyUnexpectedStatusException(u"Status '{}' not supported by session.".format(status))
```

Comment:
I think separating things out into their own Pull Requests is a standard practice that makes things easier for reviewers. If you truly don't want to, we can all make an effort here to discuss two things at once, provided everyone agrees in doing so.

Do you know where the Livy documentation regarding states is? Couldn't find it. Without it, it's hard to have any discussions, really, like which states are key (is idle the only one?).

The code was written by reading the Livy code regarding states. We started enforcing the list of known states due to a bug that would leave sessions in a non-terminal non-valid state even though there was no chance of recovery. Then, sparkmagic errored out with a non valid state reached.

I'm fine with removing the check as long as we understand the state transition graph from Livy. If you are removing the check though, please remove the variable and other places where it might be used. Do we need unit tests for the change?

I really think this should be its own Pull Request.